homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,138 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ZimiConfigEntry
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ZimiConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    async_add_entities(
+        ZimiLight(device, api)
+        for device in filter(lambda light: light.type != "dimmer", api.lights)
+    )
+    async_add_entities(
+        ZimiDimmer(device, api)
+        for device in filter(lambda light: light.type == "dimmer", api.lights)
+    )
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize a ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_should_poll = False
+        self._attr_has_entity_name = True
+        self._attr_name = None
+        self._light = light
+        self._light.subscribe(self)
+        self._attr_color_mode = ColorMode.ONOFF
+        self._attr_supported_color_modes = {ColorMode.ONOFF}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, light.identifier)},
+            name=self._light.name.strip(),
+            manufacturer=api.brand,
+            model=self._light.type,
+            suggested_area=self._light.room,
```

Comment:
will you add more platforms (switch, sensor, etc) in the future? If yes, please already add a base entity

Reply:
Done.   https://github.com/home-assistant/core/pull/129876/commits/efe9749a2eff268b4babe32122572e7787e96f4e